src/sql/upgrade-81.sql

```diff
@@ -0,0 +1,4 @@
+alter table runcommandlogs add column uuid uuid;
+update runcommandlogs set uuid = gen_random_uuid() where uuid is null;
+alter table runcommandlogs alter column uuid set not null;
+alter table runcommandlogs add constraint RunCommandLogs_uuid_unique unique(uuid);
```

Comment:
I just noticed that Postgres doesn't automatically create indexes on the _source_ end of a foreign key. A while ago I experimented with adding 120 million runcommandlogs to the table, and this is causing significant performance problems:

```
hydra=# explain analyze select * from runcommandlogs where build_id = 2;
                                                             QUERY PLAN                                                             
------------------------------------------------------------------------------------------------------------------------------------
 Gather  (cost=1000.00..1626001.73 rows=1 width=50) (actual time=3718.813..3719.702 rows=5 loops=1)
   Workers Planned: 2
   Workers Launched: 2
   ->  Parallel Seq Scan on runcommandlogs  (cost=0.00..1625001.62 rows=1 width=50) (actual time=3656.869..3717.351 rows=2 loops=3)
         Filter: (build_id = 2)
         Rows Removed by Filter: 39999999
 Planning Time: 0.251 ms
 Execution Time: 3719.737 ms
(8 rows)
```

We can see there isn't an existing index on the build_id:

```
hydra=# \d+ runcommandlogs;
                                                   Table "public.runcommandlogs"
    Column    |  Type   | Collation | Nullable |                  Default                   | Storage  | Stats target | Description 
--------------+---------+-----------+----------+--------------------------------------------+----------+--------------+-------------
 id           | integer |           | not null | nextval('runcommandlogs_id_seq'::regclass) | plain    |              | 
 job_matcher  | text    |           | not null |                                            | extended |              | 
 build_id     | integer |           | not null |                                            | plain    |              | 
 command      | text    |           | not null |                                            | extended |              | 
 start_time   | integer |           |          |                                            | plain    |              | 
 end_time     | integer |           |          |                                            | plain    |              | 
 error_number | integer |           |          |                                            | plain    |              | 
 exit_code    | integer |           |          |                                            | plain    |              | 
 signal       | integer |           |          |                                            | plain    |              | 
 core_dumped  | boolean |           |          |                                            | plain    |              | 
 uuid         | uuid    |           | not null |                                            | plain    |              | 
Indexes:
    "runcommandlogs_pkey" PRIMARY KEY, btree (id)
    "runcommandlogsuuid" btree (uuid)
Check constraints:
    "runcommandlogs_end_time_has_start_time" CHECK (end_time IS NULL OR start_time IS NOT NULL)
    "runcommandlogs_not_started_no_exit_time_no_code" CHECK (start_time IS NOT NULL OR end_time IS NULL AND error_number IS NULL AND exit_code IS NULL AND signal IS NULL AND core_dumped IS NULL)
Foreign-key constraints:
    "runcommandlogs_build_id_fkey" FOREIGN KEY (build_id) REFERENCES builds(id) ON DELETE CASCADE
Access method: heap
```

I tried adding one with:

```suggestion
alter table runcommandlogs add constraint RunCommandLogs_uuid_unique unique(uuid);
create index IndexRunCommandLogsOnBuildID on RunCommandLogs(build_id);
```

and now we're getting very efficient queries:

```
hydra=# explain analyze select * from runcommandlogs where build_id = 2;
                                                                  QUERY PLAN                                                                  
----------------------------------------------------------------------------------------------------------------------------------------------
 Index Scan using indexruncommandlogsonbuildid on runcommandlogs  (cost=0.57..4.58 rows=1 width=50) (actual time=0.037..0.039 rows=5 loops=1)
   Index Cond: (build_id = 2)
 Planning Time: 0.172 ms
 Execution Time: 0.051 ms
(4 rows)
```

Can you alter hydra.sql and this migration to add this index while we're in here?